.github/actions/verify-metrics-snapshot/action.yaml

```diff
@@ -44,17 +44,35 @@ runs:
         path: ./.metrics/cached_${{ inputs.snapshot}}
         key: ${{ env.v2_tag }}_${{ inputs.artifact_key }}
 
+    - name: Check cache and exit action
+      if: steps.cache-metrics.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        echo "::notice::No cached metrics found for comparison, skipping remaining steps"
+        exit 78 
+
     - name: Calculate diff between the snapshots 
+      id: diff-check
       shell: bash
       run: |
-        python3 -m pip install prometheus-client deepdiff
-        python3 ./scripts/e2e/compare_metrics.py --file1 ./.metrics/${{ inputs.snapshot }} --file2 ./.metrics/cached_${{ inputs.snapshot}} --output ./.metrics/diff_${{ inputs.snapshot }}
+        python3 -m pip install prometheus-client
+        if ! python3 ./scripts/e2e/compare_metrics.py --file1 ./.metrics/${{ inputs.snapshot }} --file2 ./.metrics/cached_${{ inputs.snapshot}} --output ./.metrics/diff_${{ inputs.snapshot }}; then
+          echo "diff_found=true" >> $GITHUB_ENV
```

Comment:
you shouldn't use ENV here, but the output

Reply:
example of using GITHUB_OUTPUT https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#example-masking-a-generated-output-within-a-single-job